### DISCUSSION TITLE
Should reth return SYNCING for FCU when hooks are running?

### DISCUSSION THREAD

**Discussion by ImTei** [paradigmxyz/reth#8298] [Technical discussions]
There are several open issues(https://github.com/paradigmxyz/reth/issues/7500) regarding op-reth sync.

As I understand, here's how this issue happens:
1. op-reth runs pruner periodically.
2. If op-reth gets FCU request while pruner is running, it [returns SYNCING](https://github.com/paradigmxyz/reth/blob/main/crates/consensus/beacon/src/engine/mod.rs#L363-L376).
3. op-node handles SYNCING as a critical error, because the payload is derived from L1 batches and verified, so it must be VALID.
4. op-node derivation pipeline is stopped because of a critical error.

According to the [Ethereum engine API specs](https://github.com/ethereum/execution-apis/blob/main/src/engine/paris.md#engine_forkchoiceupdatedv1), the condition of SYNCING is like this:
> {payloadStatus: {status: SYNCING, latestValidHash: null, validationError: null}, payloadId: null} if forkchoiceState.headBlockHash references an unknown payload or a payload that can't be validated because requisite data for the validation is missing

And also the spec said:
> If any of the above fails due to errors unrelated to the normal processing flow of the method, client software MUST respond with an error object.

Skipping FCU due to running hooks seems to be `errors unrelated to the normal processing flow of the method`. So I think it should return an error object instead of SYNCING, to make op-node can handle it as a temporary error.

What do you think? I'm new to reth code so I may be wrong here, if so please let me know!

**Comment by emhane** [comment]
@joshieDo , you have been working on this, right?
